packages/devtools/test/service_manager_test.dart

```diff
@@ -44,6 +44,48 @@ void main() {
       await env.tearDownEnvironment();
     });
 
+    test('invalid setBreakpoint throws exception', () async {
+      await env.setupEnvironment();
+      // Service with more than 1 registration.
+      serviceManager.registeredMethodsForService.putIfAbsent(
+        'fakeMethod',
+        () => ['registration1.fakeMethod', 'registration2.fakeMethod'],
+      );
+      expect(serviceManager.callService('fakeMethod'), throwsException);
+
+      final Completer<Object> testDone = Completer();
+      Object testError;
+      runZoned(() {
+        Future<void> asyncTestMethod() async {
+          // Service with less than 1 registration.
+          bool caughtException = false;
+          // TODO(jacobr): there is a simpler pattern for writing this I'm
+          // forgetting.
+          try {
+            await serviceManager.service.addBreakpoint(
+                serviceManager.isolateManager.selectedIsolate.id,
+                'fake-script-id',
+                1);
+          } catch (e) {
+            caughtException = true;
+          }
+          expect(caughtException, isTrue);
```

Comment:
Possibly:

```
Future breakpointResponse = serviceManager.service.addBreakpoint(...);
expect(breakpointResponse, throws);
```

I'm not sure if that handles futures; if not, there should be a similar API which does.

Reply:
Or, `await expectLater(...)`?

Reply:
used `expect(..., throwsException);` instead of try/catch.